sources/packages/web/src/components/aest/students/modals/UpdateDisabilityStatusModal.vue

```diff
@@ -0,0 +1,108 @@
+    >
+      <template #content>
+        <error-summary :errors="updateDisabilityStatusForm.errors" />
+        <p class="label-value">
+          Update the disability status of student and add note.
+        </p>
+        <v-select
+          label="Disability Status"
+          density="compact"
+          :items="disabilityStatusSelectItems"
+          v-model="formModel.disabilityStatus"
+          variant="outlined"
+          :rules="[(v) => checkNullOrEmptyRule(v, 'Disability Status')]"
+        />
+        <v-textarea
+          v-model="formModel.noteDescription"
+          variant="outlined"
+          label="Note"
+          :rules="[checkNotesLengthRule]"
+          required
+          class="mt-4"
+          hide-details="auto"
+        ></v-textarea>
+      </template>
+      <template #footer>
+        <check-permission-role :role="allowedRole">
+          <template #="{ notAllowed }">
+            <footer-buttons
+              primaryLabel="Add record now"
+              @secondaryClick="cancel"
+              @primaryClick="submit"
+              :disablePrimaryButton="notAllowed"
+            />
+          </template>
+        </check-permission-role>
+      </template>
+    </modal-dialog-base>
+  </v-form>
+</template>
+<script lang="ts">
+import { PropType, ref, reactive, defineComponent } from "vue";
+import { useModalDialog, useRules, useFormatters } from "@/composables";
+import { Role, VForm, SelectItemType, DisabilityStatus } from "@/types";
+import { UpdateDisabilityStatusAPIInDTO } from "@/services/http/dto";
+import ErrorSummary from "@/components/generic/ErrorSummary.vue";
+import ModalDialogBase from "@/components/generic/ModalDialogBase.vue";
+import CheckPermissionRole from "@/components/generic/CheckPermissionRole.vue";
+
+export default defineComponent({
+  components: { ModalDialogBase, CheckPermissionRole, ErrorSummary },
+  props: {
+    allowedRole: {
+      type: String as PropType<Role>,
+      required: true,
+    },
+  },
+  setup() {
+    const { checkNullOrEmptyRule, checkNotesLengthRule } = useRules();
+    const { disabilityStatusToDisplay } = useFormatters();
+    const { showDialog, showModal, resolvePromise } = useModalDialog<
+      UpdateDisabilityStatusAPIInDTO | boolean
+    >();
+    const updateDisabilityStatusForm = ref({} as VForm);
+    const formModel = reactive({} as UpdateDisabilityStatusAPIInDTO);
+    const disabilityStatusSelectItems = Object.keys(
+      DisabilityStatus,
+    ).map<SelectItemType>((key) => ({
+      title: disabilityStatusToDisplay(DisabilityStatus[key]),
+      value: DisabilityStatus[key],
+    }));
+
+    const submit = async () => {
+      const validationResult =
+        await updateDisabilityStatusForm.value.validate();
+      if (!validationResult.valid) {
+        return;
+      }
+      // Copying the payload, as reset is making the formModel properties null.
+      const payload = { ...formModel };
+      resolvePromise(payload);
```

Comment:
During one of the latest @sh16011993 PR he introduced the `keepModalOpen: true` for the modal `resolvePromise` to allow showing the button loading in the modal while waiting for the API response.
Either we do not have a global loading strategy we have many places where we added the out-of-box button loading at least.
Can you please consider adding the loading to the modal button (following Shashank's changes)?
For this case, the `footer-buttons` has the `processing` to be used.

Reply:
Done.